subgraph/src/fleek-nfa.ts

```diff
@@ -80,6 +82,52 @@ export function handleCollectionRoleGranted(
   entity.transactionHash = event.transaction.hash;
 
   entity.save();
+
+  if (event.params.role === 0) {
+    // Role 0 => Owner [Probably going to change this after the ACL refactor.]
+    // Should create a new CollectionOwner entity with the address from the parameters.
+    // If it already is a collection owner, should log a warning.
+
+    let collectionOwner = CollectionOwner.load(event.params.toAddress);
+
+    if (collectionOwner) {
+      // Collection Owner already exists.
+      // Print warning log message.
+      log.warning(
+        'Although Address {} is already a collection owner, a CollectionRoleGranted event was emitted that indicated the address was granted the same role, again.',
+        [event.params.toAddress.toHexString()]
+      );
```

Comment:
Shouldn't this scenario still update the stored collection owner?

Reply:
It's the same change and the check before indicates this modification has already been applied before and the current state of the subgraph includes it.